aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -972,19 +1017,58 @@ public void connect(
             final String customAuthorizer,
             final AWSIotMqttClientStatusCallback statusCallback) {
 
-        if (tokenKeyName == null || token == null || tokenSignature == null) {
+        if (tokenKeyName == null || token == null) {
             throw new IllegalArgumentException("tokenKeyName/token/tokenSignature cannot be null");
         }
 
         this.userStatusCallback = statusCallback;
 
+        final MqttConnectOptions options = new MqttConnectOptions();
+        // Set custom websocket header
+        customWebsocketHeaders.setProperty("X-Amz-CustomAuthorizer-Name", customAuthorizer);
+        customWebsocketHeaders.setProperty(tokenKeyName, token);
+        if (tokenSignature != null) {
+            customWebsocketHeaders.setProperty("X-Amz-CustomAuthorizer-Signature", tokenSignature);
+        }
+        customAuthConnect(options);
+    }
+
+    /**
+     * Initializes the MQTT session and connects to the specified MQTT server
+     * using username and password as credentials.
+     *
+     * @param username Username to authenticate client to the broker.
+     * @param password Password to authenticate client to the broker.
+     * @param statusCallback When a new MQTT session status is received, a function of callback will
+     *                       be called with new connection status.
+     */
+    public void connect(
+            final String username,
+            final String password,
+            final AWSIotMqttClientStatusCallback statusCallback) {
+
+        if (username == null || password == null) {
+            throw new IllegalArgumentException("username and/or password cannot be null.");
+        }
+
+        this.username = username;
+        this.password = password;
+        this.userStatusCallback = statusCallback;
```

Comment:
Same here, do you need to cache this on the class? Or can you just pass it as an argument to `customerAuthConnect(options, statusCallback)`? 

Currently, have you have `null` guards on the `username` and `password`, but not on this `statusCallback`. Is this `connect(...)` method expected to work, if the user passes `null` for the callback? If not, you could add a guard for it too.

Reply:
`statuscallback` is allowed to be null and we notify the user only if a non-null status callbacks is provided. This callback is being invoked for status change notification from connects, disconnect and reconnect and is therefore class-scoped.